includes/REST_API/Publisher_Logos_Controller.php

```diff
@@ -0,0 +1,428 @@
+
+		if ( is_wp_error( $post ) ) {
+			return $post;
+		}
+
+		if ( $request['active'] ) {
+			update_option( Settings::SETTING_NAME_ACTIVE_PUBLISHER_LOGO, $post->ID );
+		}
+
+		return $this->prepare_item_for_response( $post, $request );
+	}
+
+	/**
+	 * Get an existing publisher logo's post object, if valid.
+	 *
+	 * @since 1.12.0
+	 *
+	 * @param int $id Supplied ID.
+	 * @return WP_Post|WP_Error Post object if ID is valid, WP_Error otherwise.
+	 */
+	protected function get_publisher_logo( $id ) {
+		$publisher_logos = $this->filter_publisher_logos( (array) get_option( Settings::SETTING_NAME_PUBLISHER_LOGOS, [] ) );
+
+		$post = get_post( $id );
+
+		if ( ! $post || ! in_array( $post->ID, $publisher_logos, true ) ) {
+			return new WP_Error(
+				'rest_invalid_id',
+				__( 'Invalid ID', 'web-stories' ),
+				[ 'status' => 400 ]
+			);
+		}
+
+		return $post;
+	}
+
+	/**
+	 * Prepares a single publisher logo output for response.
+	 *
+	 * @since 1.12.0
+	 *
+	 * @param WP_Post         $post    Post object.
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response Response object.
+	 */
+	public function prepare_item_for_response( $post, $request ): WP_REST_Response {
+		$fields = $this->get_fields_for_response( $request );
+
+		// Base fields for every post.
+		$data = [];
+
+		if ( rest_is_field_included( 'id', $fields ) ) {
+			$data['id'] = $post->ID;
+		}
+
+		if ( rest_is_field_included( 'title', $fields ) ) {
+			$data['title'] = get_the_title( $post->ID );
+		}
+
+		if ( rest_is_field_included( 'active', $fields ) ) {
+			$active_publisher_logo_id = absint( get_option( Settings::SETTING_NAME_ACTIVE_PUBLISHER_LOGO ) );
+			$data['active']           = $post->ID === $active_publisher_logo_id;
+		}
+
+		if ( in_array( 'url', $fields, true ) ) {
+			$data['url'] = wp_get_attachment_url( $post->ID );
+		}
+
+		/**
+		 * Wrapped response object.
+		 *
+		 * @var WP_REST_Response $response Response object.
+		 */
+		$response = rest_ensure_response( $data );
+
+		$links = $this->prepare_links( $post );
+		$response->add_links( $links );
+
+		return $response;
+	}
+
+	/**
+	 * Filters publisher logos to remove non-existent or invalid ones.
+	 *
+	 * @param int[] $publisher_logos List of publisher logos.
+	 * @return int[] Filtered list of publisher logos.
+	 */
+	protected function filter_publisher_logos( $publisher_logos ): array {
+		return array_filter( $publisher_logos, 'wp_attachment_is_image' );
+	}
+
+	/**
+	 * Prepares links for the request.
+	 *
+	 * @since 1.12.0
+	 *
+	 * @param WP_Post $post Post object.
+	 * @return array Links for the given post.
+	 */
+	protected function prepare_links( $post ): array {
+		$base = sprintf( '%s/%s', $this->namespace, $this->rest_base );
+
+		// Entity meta.
+		$links = [
+			'self'       => [
+				'href' => rest_url( trailingslashit( $base ) . $post->ID ),
+			],
+			'collection' => [
+				'href' => rest_url( $base ),
+			],
```

Comment:
```suggestion
			],
			'https://api.w.org/publisherlogo' => [
				'href'       => rest_url( sprintf( '%s/%s/%s', $this->namespace, 'media', $post->ID ) ),
				'embeddable' => true,
			],
```

Does it make sense to have the attachment rest embedded here?

Reply:
I thought about it, but I didn't see any benefits in doing so.

The current solution is super lean, with just like 4 fields that we need.

It would be the more RESTful way to do it, but in reality we would just end up using `?_fields` to request as few fields as needed and adding some custom JS function to get the data into the same shape as we have today. So just a lot of extra code to get to the same result.

Reply:
Is it worth adding the link so we could use later down the line?

Reply:
If we should ever need this down the line, it can be easily added then